includes/signature/class-http-message-signature.php

```diff
@@ -22,6 +22,50 @@
  */
 class Http_Message_Signature implements Signature_Standard {
 
+	/**
+	 * Generate RFC-9421 compliant Signature-Input and Signature headers for an outgoing HTTP request.
+	 *
+	 * @param array  $args The request arguments.
+	 * @param string $url  The request URL.
+	 * @return array Request arguments with signature headers.
+	 */
+	public function sign( $args, $url ) {
```

Comment:
do we need some kind of `$args` validation? or at least set defaults with `wp_parse_args` to not produce any "array key does" not exist errors?

Reply:
`Signature::sign_request()` is almost like a filter callback that runs before the request args get passed to `wp_remote_request()`. Actually, now that key_id and private_key are passed as part of the request args, we could turn it into a filter callback for `http_request_args`.

With that in mind, I think it would be good to not validate and return errors, but maybe just bail if we find that key_id and private_key are not available?
The date header we could add as a default parameter, and potentially also the request method. Although if we do end up making it a filter, the method will always be set.

Long story short: Yes, that's a good idea!

Reply:
Fixed in [5573da5](https://github.com/Automattic/wordpress-activitypub/pull/1858/commits/5573da5f9fa88c16bb41ffee576e4476e13f574e).